### PR TITLE
agnhost: Try both in-cluster and external discovery

### PR DESCRIPTION
**Pull request by mtaufen**
The conformance test for ServiceAccountIssuerDiscovery is currently
configured with --in-cluster-discovery, which only supports token
validation against in-cluster endpoints. Many cloud providers provide
their own, external endpoints for OIDC discovery, and because the iss
claim in tokens will point to these endpoints, but the client in this
test only trusts the Cluster CA, it will fail to connect to the external
discovery endpoints when validating the token.

To ensure that the conformance test at least supports scenario where
both the discovery doc endpoint and JWKS endpoint are cluster-local and
the scenario where both endpoints are cluster-external, this PR has the
test try both and requires at least one to pass.

Caveat: The test still won't support a configuration where one
endpoint is cluster-local and the other is external. We don't yet have
evidence that this is a configuration that is used in practice, so this
initial hotfix will at least fix the conformance test for the "both
external" configuration we know providers already use. Note that if one
endpoint is cluster-local, and the other is cluster-external, tokens can
still only be validated in-cluster, because both endpoints must be
accessible to Relying Parties that validate tokens.

#### What type of PR is this?
/kind bug
/kind failing-test

#### Does this PR introduce a user-facing change?
```release-note
Resolves an issue with the "ServiceAccountIssuerDiscovery should support OIDC discovery" conformance test failing on clusters which are configured with issuers outside the cluster
```

Hold since I still need to test this for GKE to ensure it can succeed in that environment:
/hold

/assign @liggitt

### CODE REVIEW

**Comment by liggitt** [comment]
need to bump test/images/agnhost/VERSION as well

**Comment by mtaufen** [comment]
Bumped.

**Comment by liggitt** [comment]
looks like build/dependencies.yaml is unhappy, and will want you to update `test/images/agnhost/agnhost.go` as well

File: test/images/agnhost/openidmetadata/openidmetadata.go [hunk 1]
```diff
@@ -48,32 +49,63 @@ var CmdTestServiceAccountIssuerDiscovery = &cobra.Command{
 }
 
 var (
-	tokenPath          string
-	audience           string
-	inClusterDiscovery bool
+	tokenPath                    string
+	audience                     string
+	deprecatedInClusterDiscovery bool
```
Comment (liggitt) [inline test/images/agnhost/openidmetadata/openidmetadata.go:54]: I'd suggest dropping this rather than keeping it registered and ignored

**Comment by liggitt** [comment]
looks reasonable otherwise

**Comment by mtaufen** [comment]
/hold cancel

I've tested this on GKE and it works with our cluster-external endpoints.

File: test/images/agnhost/openidmetadata/openidmetadata.go (hunk 1 above)
Comment (mtaufen) [reply test/images/agnhost/openidmetadata/openidmetadata.go]: Done

**Comment by liggitt** [comment]
/lgtm
/approve

post-merge, follow https://github.com/kubernetes/kubernetes/blob/master/test/images/README.md#promoting-images to update https://github.com/kubernetes/k8s.io/blob/main/k8s.gcr.io/images/k8s-staging-e2e-test-images/images.yaml, then open a follow-up to bump the manifest version used to 2.32

**Comment by mtaufen** [comment]
sg, thanks!

**Pull request merged**